crates/ide/src/navigation_target.rs

```diff
@@ -40,7 +40,7 @@ pub struct NavigationTarget {
     ///
     /// Clients should place the cursor on this range when navigating to this target.
     pub focus_range: Option<TextRange>,
-    pub name: SmolStr,
+    pub name: NavigationTargetName,
```

Comment:
Thinking a bit more about this. Showing the aliased path is redundant as the client can already render the file path anyways, so let's instead keep the `name` field as is, add another `alias: Option<SmolStr>` field (with a fitting doc comment) and use those appropriately. Then we just render `<alias> (alias <original name>)`

Then we don't have to deal with the canonical paths either

Reply:
I will follow the format `<alias> (alias <original name>)` but I think that aliases come in second order when compared to their original names. Meaning that it would make more sense if we rendered <original> (alias <alias>). It is equally intuitive to search and it looks more orderly. In addition to this the user can e.g list all the aliases by 

1. Finding the original name by typing the alias.
2. Making a new search by typing the original name and thus getting a list like 

```
+ <ORIGINAL> ( alias <ALIAS_1> )
+ <ORIGINAL> ( alias <ALIAS_2> )
```